tests/datasets/test_geo.py

```diff
@@ -368,6 +377,143 @@ def test_no_all_bands(self) -> None:
             CustomSentinelDataset(root, bands=bands, transforms=transforms, cache=cache)
 
 
+class TestVirtualFilesystems:
+    @pytest.mark.parametrize(
+        'temp_archive', [os.path.join('tests', 'data', 'vector')], indirect=True
+    )
+    def test_zipped_file(self, temp_archive: tuple[str, str]) -> None:
+        _, dir_zipped = temp_archive
+        filename = 'vector_2024.geojson'
+
+        # Using '!' as separator between archive and file as per Apache Commons VFS
+        # https://rasterio.readthedocs.io/en/latest/topics/vsi.html
+        specific_file_zipped = f'{dir_zipped}!{filename}'
+
+        vsi_path = f'zip://{specific_file_zipped}'
+        if sys.platform == 'win32':
+            vsi_path = os.path.join('zip://D:', specific_file_zipped)
+
+        files_found = CustomGeoDataset(paths=vsi_path).files
+        assert len(files_found) == 1
+        file = str(files_found[0])
+        assert file.endswith(filename)
+
+    @pytest.mark.parametrize(
+        'temp_archive',
+        [
+            os.path.join(
+                'tests',
+                'data',
+                'sentinel2',
+                'S2A_MSIL2A_20220414T110751_N0400_R108_T26EMU_20220414T165533.SAFE',
+            )
+        ],
+        indirect=True,
+    )
+    def test_zipped_specific_file_dir(self, temp_archive: tuple[str, str]) -> None:
+        dir_not_zipped, dir_zipped = temp_archive
+
+        # Listing, instead of hardcoding filepath
+        filename_glob = Sentinel2.filename_glob.format('B02')
+        specific_file_not_zipped = list_directory_recursive(
+            dir_not_zipped, filename_glob
+        )[0]
+        filepath_within_dir = specific_file_not_zipped.replace(dir_not_zipped, '')
+
+        specific_file_zipped = f'zip://{dir_zipped}!{filepath_within_dir}'
+        if sys.platform == 'win32':
+            specific_file_zipped = os.path.join(
+                'zip://D:', '{dir_zipped}!{filepath_within_dir}'
+            )
```

Comment:
Tried my luck looking at [this test in fiona](https://github.com/Toblerity/Fiona/blob/908bf4f8a07d36ec1d55ba45f6550a3699520b6f/tests/test__path.py#L8), but I guess this is only required when using abspath.

Would I need to specify absolute path? My intuition tells me no

Reply:
Definitely no need to test abs path. We aren't trying to test fiona, we're trying to test TorchGeo.